flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemTableSink.java

```diff
@@ -373,16 +374,16 @@ public DynamicTableSource.DataStructureConverter createDataStructureConverter(
         };
     }
 
-    private Path toStagingPath() {
+    @VisibleForTesting
+    Path toStagingPath() {
         // Add a random UUID to prevent multiple sinks from sharing the same staging dir.
         // Please see FLINK-29114 for more details
         Path stagingDir =
                 new Path(path, ".staging_" + UUID.randomUUID() + System.currentTimeMillis());
         try {
             FileSystem fs = stagingDir.getFileSystem();
             Preconditions.checkState(
-                    fs.exists(stagingDir) || fs.mkdirs(stagingDir),
-                    "Failed to create staging dir " + stagingDir);
+                    fs.mkdirs(stagingDir), "Failed to create staging dir " + stagingDir);
```

Comment:
The Precondition should be updated as well: I guess we should allow existing folders if we really want to have unique folders here. :thinking:

Reply:
Can you elaborate further? I'm a little bit confused about this. 
If we want to have unique folders, then encountering an already existent dir means sth. goes wrong. Shall we allow this?

Reply:
The precondition checks first if the `stagingDir` exists and only calls `mkdirs` if the exist check is false. What we want is that the `stagingDir` does not exist with `mkdirs` returning `true` because at least one subfolder of the `stagingDir`'s path needed to be created.

Reply:
Thanks for the clarification. Got your point. +1